File: web/src/main/java/ru/protei/portal/ui/web/client/model/event/EventBusEventAuthLoginDone.java
package ru.protei.portal.ui.web.client.model.event;

import jsinterop.annotations.JsOverlay;

public class EventBusEventAuthLoginDone extends EventBusEvent {
    protected EventBusEventAuthLoginDone() {}
    public static String type = "@app-portal/auth-login-done";

    public static native EventBusEventAuthLoginDone create(double personId, double loginId) /*-{
        return {
            type: @ru.protei.portal.ui.web.client.model.event.EventBusEventAuthLoginDone::type,
            source: @ru.protei.portal.ui.web.client.model.event.EventBusSource::SourcePortalUiGwt,
            payload: {
                personId: personId,
                loginId: loginId
            }
        };
    }-*/;

    @JsOverlay public final native Long getPersonId() /*-{ return this.payload.personId; }-*/;
    @JsOverlay public final native Long getLoginId() /*-{ return this.payload.loginId; }-*/;
}

File: web/packages/common/common/src/js/model/eventbus/event/auth/login/EventBusEventAuthLoginDone.ts
import { EventBusEvent, EventSource } from "@protei-libs/eventbus"
import { EventBusSourcePortalUi } from "../../../source/EventBusSource"
import { PersonId, UserLoginId } from "../../../../entity"

export const EventBusEventAuthLoginDoneType = "@app-portal/auth-login-done"

export type EventBusEventAuthLoginDone = ReturnType<typeof eventBusEventAuthLoginDone>

export type EventBusEventAuthLoginDonePayload = {
  personId: PersonId
  loginId: UserLoginId
}

export const eventBusEventAuthLoginDone = (
  payload: EventBusEventAuthLoginDonePayload,
  options?: { source?: EventSource },
): EventBusEvent<EventBusEventAuthLoginDonePayload> =>
  <const>{
    type: EventBusEventAuthLoginDoneType,
    source: options?.source ?? EventBusSourcePortalUi,
    payload: payload,
  }
